cpp/tests/stream_compaction/drop_duplicates_tests.cpp

```diff
@@ -14,254 +14,245 @@
+  // Keep first of duplicate
+  // The expected table would be sorted in ascending order with respect to keys
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col1_first{{5, 3, 5, 8, 5}};
+  cudf::test::fixed_width_column_wrapper<float> exp_col2_first{{4, 3, 4, 9, 4}};
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col1_key_first{{20, 20, 19, 21, 9}};
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col2_key_first{{19, 20, 20, 9, 21}};
+  cudf::table_view expected_first{
+    {exp_col1_first, exp_col2_first, exp_col1_key_first, exp_col2_key_first}};
 
-  cudf::size_type expected = 4;
-  EXPECT_EQ(expected,
-            cudf::distinct_count(input_col, null_policy::INCLUDE, nan_policy::NAN_IS_NULL));
-}
+  auto got_first = drop_duplicates(input, keys, cudf::duplicate_keep_option::KEEP_FIRST);
 
-TEST_F(DistinctCount, NansAsNullWithIgnoreNull)
-{
-  using T = float;
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_first, got_first->view());
 
-  std::vector<T> input               = {1, 3, NAN, 70, 31};
-  std::vector<cudf::size_type> valid = {1, 1, 1, 0, 1};
+  // keep last of duplicate
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col1_last{{4, 3, 5, 8, 5}};
+  cudf::test::fixed_width_column_wrapper<float> exp_col2_last{{5, 3, 4, 9, 4}};
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col1_key_last{{20, 20, 19, 21, 9}};
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col2_key_last{{19, 20, 20, 9, 21}};
+  cudf::table_view expected_last{
+    {exp_col1_last, exp_col2_last, exp_col1_key_last, exp_col2_key_last}};
 
-  cudf::test::fixed_width_column_wrapper<T> input_col{input.begin(), input.end(), valid.begin()};
+  auto got_last = drop_duplicates(input, keys, cudf::duplicate_keep_option::KEEP_LAST);
 
-  cudf::size_type expected = 3;
-  EXPECT_EQ(expected,
-            cudf::distinct_count(input_col, null_policy::EXCLUDE, nan_policy::NAN_IS_NULL));
-}
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_last, got_last->view());
 
-TEST_F(DistinctCount, EmptyColumn)
-{
-  using T = float;
+  // Keep unique
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col1_unique{{3, 5, 8, 5}};
+  cudf::test::fixed_width_column_wrapper<float> exp_col2_unique{{3, 4, 9, 4}};
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col1_key_unique{{20, 19, 21, 9}};
+  cudf::test::fixed_width_column_wrapper<int32_t> exp_col2_key_unique{{20, 20, 9, 21}};
+  cudf::table_view expected_unique{
+    {exp_col1_unique, exp_col2_unique, exp_col1_key_unique, exp_col2_key_unique}};
 
-  cudf::test::fixed_width_column_wrapper<T> input_col{};
+  auto got_unique = drop_duplicates(input, keys, cudf::duplicate_keep_option::KEEP_NONE);
 
-  cudf::size_type expected = 0;
-  EXPECT_EQ(expected,
-            cudf::distinct_count(input_col, null_policy::EXCLUDE, nan_policy::NAN_IS_NULL));
+  CUDF_TEST_EXPECT_TABLES_EQUAL(expected_unique, got_unique->view());
 }
 
-TEST_F(DistinctCount, StringColumnWithNull)
+TEST_F(DropDuplicate, WithNull)
```

Comment:
Split up these tests. The variations below appear to act on separate data and have separate assertions, so there's no clear tie between them. Also applies to other tests updated in this PR.